src/ng2-facebook-sdk.ts

```diff
@@ -25,15 +28,19 @@ export class FacebookService {
     api(path: string, method: FacebookApiMethod = 'get', params: any = {}): Promise<any> {
         return new Promise<any>(
             (resolve, reject) => {
-                FB.api(path, method, params, (response: any) => {
-                    if(!response){
-                        reject();
-                    }else if(response.error){
-                        reject(response.error);
-                    }else{
-                        resolve(response);
-                    }
-                });
+                try {
+                    FB.api(path, method, params, (response: any) => {
+                        if(!response){
+                            reject();
+                        }else if(response.error){
+                            reject(response.error);
+                        }else{
+                            resolve(response);
+                        }
+                    });
+                } catch (err) {
+                    console.error(err);
```

Comment:
Since we already have a promise here, we could reject the promise with the given error instead of logging it to the console.

Please apply this change to the other promises below.